snmp/datadog_checks/snmp/config.py

```diff
@@ -200,41 +236,82 @@ def get_table_symbols(mib, table):
+                            object_type = hlapi.ObjectType(identity)
                         except Exception as e:
                             warning("Can't generate MIB object for variable : %s\nException: %s", metric, e)
-                    if 'metric_tags' in metric:
-                        for metric_tag in metric['metric_tags']:
-                            if not ('tag' in metric_tag and ('index' in metric_tag or 'column' in metric_tag)):
+                        else:
+                            if 'table' in metric_tag:
+                                tag_symbols = get_table_symbols(mib, metric_tag['table'])
+                                tag_symbols.append(object_type)
+                            elif mib != metric['MIB']:
                                 raise ConfigurationError(
-                                    'When specifying metric tags, you must specify a tag, and an index or column'
+                                    'When tagging from a different MIB, the table must be specified'
                                 )
-                            if 'column' in metric_tag:
-                                # In case it's a column, we need to query it as well
-                                mib = metric_tag.get('MIB', metric['MIB'])
-                                try:
-                                    object_type = hlapi.ObjectType(hlapi.ObjectIdentity(mib, metric_tag['column']))
-                                except Exception as e:
-                                    warning("Can't generate MIB object for variable : %s\nException: %s", metric, e)
-                                else:
-                                    if 'table' in metric_tag:
-                                        tag_symbols = get_table_symbols(mib, metric_tag['table'])
-                                        tag_symbols.append(object_type)
-                                    elif mib != metric['MIB']:
-                                        raise ConfigurationError(
-                                            'When tagging from a different MIB, the table must be specified'
-                                        )
-                                    else:
-                                        symbols.append(object_type)
-
+                            else:
+                                symbols.append(object_type)
+                    elif 'index' in metric_tag:
+                        index_tags.append((tag_key, metric_tag['index']))
+                        if 'mapping' in metric_tag:
+                            # Need to do manual resolution
+                            for symbol in metric['symbols']:
+                                self._resolver.register_index(
+                                    symbol['name'], metric_tag['index'], metric_tag['mapping']
+                                )
+                for symbol in metric['symbols']:
+                    if isinstance(symbol, dict):
```

Comment:
Looks like this logic is repeated multiple times, there is maybe a way to handle this in a generic way.

```
                        if isinstance(XXXX, dict):
                            self._resolver.register(to_oid_tuple(to_query['OID']), to_query['name'])
                            parsed_metric_name = to_query['name']
                        else:
                            parsed_metric_name = to_query['name']
```

Ideally, if possible, the resolver logic should be generic and done at one place only.

Reply:
Good idea, I tried something.